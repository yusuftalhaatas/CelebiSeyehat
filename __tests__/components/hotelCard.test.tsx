import React from 'react';
import {render} from '@testing-library/react-native';
import HotelCard from '../../src/components/hotelCard';

describe('components/HotelCard', () => {
  test('renders correctly', () => {
    const hotelData = {
      firmName: 'Test Hotel',
      city: 'Test City',
      firmType: 'Hotel',
      roomNo: '101',
      date: '2023-05-15',
      name: 'Test Name',
    };

    const {getByText, toJSON} = render(<HotelCard {...hotelData} />);

    // snapshot test
    expect(toJSON()).toMatchSnapshot();

    // Check if all fields are displayed correctly
    expect(getByText(`Name: ${hotelData.name}`)).toBeDefined();
    expect(getByText(`Hotel Name: ${hotelData.firmName}`)).toBeDefined();
    expect(getByText(`Date: ${hotelData.date}`)).toBeDefined();
    expect(getByText(`Room No: ${hotelData.roomNo}`)).toBeDefined();
    expect(getByText(`City: ${hotelData.city}`)).toBeDefined();
  });
});
