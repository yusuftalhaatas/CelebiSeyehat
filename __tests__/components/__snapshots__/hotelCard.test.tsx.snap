// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/HotelCard renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#00ccff",
      "borderColor": "black",
      "borderRadius": 10,
      "borderWidth": 1,
      "marginHorizontal": 20,
      "marginVertical": 10,
    }
  }
>
  <Text
    style={
      {
        "backgroundColor": "#ffffff",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontSize": 17,
        "fontWeight": "bold",
        "margin": 5,
        "padding": 5,
      }
    }
  >
    Name: 
    Test Name
  </Text>
  <Text
    style={
      {
        "backgroundColor": "#ffffff",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontSize": 17,
        "fontWeight": "bold",
        "margin": 5,
        "padding": 5,
      }
    }
  >
    Hotel Name: 
    Test Hotel
  </Text>
  <Text
    style={
      {
        "backgroundColor": "#ffffff",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontSize": 17,
        "fontWeight": "bold",
        "margin": 5,
        "padding": 5,
      }
    }
  >
    Date: 
    2023-05-15
  </Text>
  <Text
    style={
      {
        "backgroundColor": "#ffffff",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontSize": 17,
        "fontWeight": "bold",
        "margin": 5,
        "padding": 5,
      }
    }
  >
    Room No: 
    101
  </Text>
  <Text
    style={
      {
        "backgroundColor": "#ffffff",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontSize": 17,
        "fontWeight": "bold",
        "margin": 5,
        "padding": 5,
      }
    }
  >
    City: 
    Test City
  </Text>
</View>
`;
